gnupg/_meta.py

```diff
@@ -652,7 +652,11 @@ def _read_response(self, stream, result):
 
                 # Log gpg's userland messages at our own levels:
                 if keyword.upper().startswith("WARNING"):
-                    log.warn("%s" % value)
+                    # Silence warnings from gpg we're supposed to ignore
+                    ignore = (self.ignore_homedir_permissions
+                        and 'unsafe ownership on homedir' in value)
```

Comment:
I'd vaguely prefer to keep a list of warnings to ignore and check value contains any of them, rather than inline every edge case check, but I can make that change after merging yours.